packages/ia-wayback-search/src/ia-wayback-search.js

```diff
@@ -0,0 +1,80 @@
+import { html, LitElement } from 'lit-element';
+import waybackSearchCSS from './styles/wayback-search';
+import searchIcon from './icon-search';
+
+class WaybackSearch extends LitElement {
+  static get styles() {
+    return waybackSearchCSS;
+  }
+
+  static get properties() {
+    return {
+      locationHandler: { type: Object },
+      waybackPagesArchived: { type: String },
+      waybackHost: { type: String },
+    };
+  }
+
+  constructor() {
+    super();
+    this.waybackPagesArchived = '';
+    this.waybackHost = 'archive.org';
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const query = e.target.querySelector('#url').value;
+    this.emitWaybackSearchSubmitted(query);
+    this.locationHandler.submitCallback(`https://${this.waybackHost}/web/*/${query}`);
```

Comment:
_Suggestion_ I would try to decouple the UI from the actual event that takes place when a search is submitted. For instance, in petabox we might want the URL redirect to happen, but in offshoot, it might cause something else like an XHR request at a different URL.

You could have this component just emit the `waybackSearchSubmitted` event like it already does, then in the consumer just listen for that event and do whatever is appropriate for that consumer (ie a `window.location` in petabox).

Reply:
That was my initial suggestion in our meeting, but you mentioned passing in an object with a method that could be mocked in tests.

Reply:
You can still take this approach passing in the handler, but I wouldn't have the component dictate the URL that it goes to.. the handler could be more generic like
```
searchHandler.performSearch(term);
```
Then the one passed in can do anything with it. Either approach works here... you're just delegating that part of the responsibility up a level.

Reply:
Ah, so you're wanting to pull the URL building out. That makes more sense.